Jint/Runtime/TypeConverter.cs

```diff
@@ -522,6 +525,25 @@ public static void CheckObjectCoercible(Engine engine, JsValue o)
                     matchingByParameterCount = matchingByParameterCount ?? new List<Tuple<T, JsValue[]>>();
                     matchingByParameterCount.Add(new Tuple<T, JsValue[]>(m, arguments));
                 }
+                else if (parameterInfos.Length > arguments.Length)
+                {
+                    // check if we got enough default values to provide all parameters (or more in case some default values are provided/overwritten)
+                    var defaultValuesCount = parameterInfos.Count(x => x.HasDefaultValue);
```

Comment:
We try to avoid usage of LINQ, could you change this to old fashioned `foreach` that increments counter, so we don't have single LINQ usage for just this.

Reply:
Done. But would you elaborate on the non-LINQ usage? Performance issues?

Reply:
Yes, LINQ is always slower and we try to avoid it in Jint.